ts/features/bonus/siciliaVola/components/VoucherInformationComponent.tsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { Image, StyleSheet } from "react-native";
+import { View } from "native-base";
+import I18n from "../../../../i18n";
+import { H5 } from "../../../../components/core/typography/H5";
+import { H3 } from "../../../../components/core/typography/H3";
+import CopyButtonComponent from "../../../../components/CopyButtonComponent";
+
+type Props = {
+  voucherCode: string;
+  qrCode: string;
+  barCode: string;
+};
+
+const styles = StyleSheet.create({
+  qrCode: {
+    width: 249,
+    height: 249
+  },
+  barCode: {
+    width: 249,
+    height: 70
+  },
+  centerContentRow: {
+    flex: 1,
+    flexDirection: "row",
+    justifyContent: "center"
+  }
+});
+
+const VoucherInformationComponent = (props: Props): React.ReactElement => (
+  <>
+    <H5 weight={"Regular"}>
+      {I18n.t("bonus.sv.components.voucherVisualDetails.uniqueCode")}
+    </H5>
+    <View spacer small />
+    <View
+      style={{
+        flex: 1,
+        flexDirection: "row"
+      }}
+    >
+      <H3 color={"bluegreyDark"}>{props.voucherCode}</H3>
+      <View hspacer />
+      <CopyButtonComponent
+        textToCopy={props.voucherCode}
+        onPressWithGestureHandler={true}
+      />
+    </View>
+    <View spacer />
+    <H5 weight={"Regular"}>
+      {I18n.t("bonus.sv.components.voucherVisualDetails.qrCode")}
+    </H5>
+    <View style={styles.centerContentRow}>
+      <Image
+        accessible={true}
+        accessibilityRole={"image"}
+        accessibilityLabel={I18n.t(
+          "bonus.sv.components.voucherVisualDetails.qrCode"
+        )}
+        source={{
+          uri: `data:image/png;base64,${props.qrCode}`
```

Comment:
Since we are, what do you think about a refactoring?

We could create an `util/image.ts` file and put it the uri building, then replace the local ones (there are few in the code base) with this

```typescript
const withBase64Uri = (imageBase64: string, format : "png" | "jpg" = "png") => `data:image/${format};base64,${imageBase64}`;
```

Reply:
Sure, created the util and refactored the component code in [71397aa](https://github.com/pagopa/io-app/commit/71397aa947fd37d5779da9c98a47146c5f3f9a66).
Refactored also the other points in which the uri was hardcoded in [7bd4e7a](https://github.com/pagopa/io-app/commit/7bd4e7ab6bd6eab87cb2de6ea0763a5d8e7ab824)